File: Arrays/Java/Easy/FindDisappearedNumbers.java

/*

Arrays Easy
https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/
*/
import java.util.*;

public class FindDisappearedNumbers {

    public static void main(String[] Args) {

        System.out.println("Problem: Find disappeared numbers");
        int[] arrayA = { 4, 3, 2, 7, 8, 2, 3, 1 };

        printList(findDisappeared(arrayA));

    }

    public static List<Integer> findDisappeared(int[] nums) {

        List<Integer> missingNums = new ArrayList<>();

        for (int i = 0; i < nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            System.out.print(index);

            if (nums[index] > 0) {
                nums[index] = -nums[index];
            }

        }

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                missingNums.add(i + 1);
            }

        }

        return missingNums;

    }

    public static void print(int[] A) {

        for (int a : A) {
            System.out.print(a + " ");
        }
        System.out.println();

    }

    public static void printList(List<Integer> value) {

        System.out.println(Arrays.toString(value.toArray()));

    }

}

File: Arrays/Python/FindDisappearedNumbers.py
def findDisappearedNumbersBruteForce(nums):
        return_list = list(range(1,len(nums)+1))
        for i in set(nums):
            try:
                return_list.remove(i)
            except:
                continue
        return return_list

def findDisappearedNumbers(self,nums):
    return_list = []
    for i in range(len(nums)):
        index = abs(nums[i])-1
        if (nums[index] > 0):
            nums[index]*=-1
                
    for i in range(len(nums)):
        if(nums[i]>0):
            return_list.append(i+1)
    return return_list